cloudinit/sources/helpers/azure.py

```diff
@@ -48,6 +48,37 @@ def impl(*args, **kwargs):
     return impl
 
 
+def byte_swapped(previous_id, current_id):
```

Comment:
The naming of this function is misleading to what it does.
Can we make this something like:
def is_byte_swapped(previous_id, current_id):

And can we have this return a bool (instead of None, versus previous_id)? It just adds complexity to the function that feels unnecessary.

Reply:
At first, I was going to name it is_byte_swapped but in some places it needs to return a string.

Reply:
It doesn't need to return a string if you made it just return bool, since there is only one functional callsite, (besides your tests) you can change the callsite in DataSourceAzure to the following:

return previous if is_byte_swapped(previous, iid) else iid

Reply:
@AOhassan  can we please make this is_byte_swapped. It makes a bit more sense to me as an explicit behavior.  Otherwise we have to always look at the implementation of byte_swapped ot see that it is  returning previous_iid if current_id is swapped. Here's a patch https://paste.ubuntu.com/p/QtXgxcnwC6/